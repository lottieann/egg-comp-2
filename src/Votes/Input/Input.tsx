import * as React from 'react'

import styled from 'styled-components'

export interface InputProps {
  label: string;
  placeholder: string;
  type: string;
  required?: boolean;
  onChange: (event) => void;
  name: string;
  value?: string;
}

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  margin: 20px;
`

const StyledLabel = styled.label`
  font-family: Helvetica;
  margin: 5px;
`

const StyledInput = styled.input`
  font-family: Helvetica;
`

export const Input = (props: InputProps) => {
  return (
    <Wrapper>
      <StyledLabel>{props.label}</StyledLabel>
      <StyledInput
        value={props.value}
        placeholder={props.placeholder}
        type={props.type}
        required={props.required}
        onChange={props.onChange}
        name={props.name}
      />
    </Wrapper>
  )
}
