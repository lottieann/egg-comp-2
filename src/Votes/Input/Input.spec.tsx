import * as React from 'react'

import 'jest-styled-components'

import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import { Input, InputProps } from './Input'

const setup = (props: InputProps) => {
  return <Input {...props} />
}

describe('Input', () => {
  it('should render Input correctly', () => {
    const { container } = render(
      setup({
        label: 'name',
        placeholder: 'Enter your name',
        type: 'text',
      }),
    )
    expect(container).toMatchSnapshot()
  })
})
