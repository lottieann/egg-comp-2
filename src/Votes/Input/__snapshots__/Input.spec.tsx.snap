// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render Input correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 20px;
}

.c1 {
  font-family: Helvetica;
  margin: 5px;
}

.c2 {
  font-family: Helvetica;
}

<div>
  <div
    class="c0"
  >
    <label
      class="c1"
    >
      name
    </label>
    <input
      class="c2"
      placeholder="Enter your name"
      type="text"
      value=""
    />
  </div>
</div>
`;
